src/sp/it/pl/audio/tagging/Ext.kt

```diff
@@ -35,10 +35,10 @@ val Tag.ratingMin: Int get() = 0
 fun Tag.clipRating(v: Double): Double = clip(ratingMin.toDouble(), v, ratingMax.toDouble())
 
 /** @return audio file or error if fails */
-fun File.readAudioFile(): Try<AudioFile, Void> {
+fun File.readAudioFile(): Try<AudioFile, Exception> {
     val onError = { e: Exception ->
         logger.error(e) { "Reading metadata failed for file $this" }
```

Comment:
The exception is returned so the caller can get the specifics of the error,

The logging must always happen when error occurs, immediately. Sometimes an exception is not an error (like when validating user input), though usually it is. In this case it is, this it is logged.

Reply:
> In this case it is, thus it is logged.
You can't ever say this definitively. What if in the situation this method is called it is not sure whether it is an AudioFile and failing is not an exceptional case? Then it shouldn't be logged. That's why I am strongly proposing leaving the responsibility to the caller, to avoid cluttered logs.

Reply:
I see. That is a sound argument. We should continue this elsewhere.

Because it is unrelated (as the logging has already been there from the start) Im considering this resolved and proceed withthe merge.